src/sinks/azure_blob.rs

```diff
@@ -0,0 +1,832 @@
+        assert_eq!(
+            blob.properties.content_type,
+            String::from("application/octet-stream")
+        );
+        let expected = events
+            .iter()
+            .map(|event| serde_json::to_string(&event.as_log().all_fields()).unwrap())
+            .collect::<Vec<_>>();
+        assert_eq!(expected, blob_lines);
+    }
+
+    #[tokio::test]
+    async fn azure_blob_rotate_files_after_the_buffer_size_is_reached() {
+        let blob_prefix = String::from("lines-rotate/into/blob/");
+        let config = AzureBlobSinkConfig::new_emulator().await;
+        let config = AzureBlobSinkConfig {
+            blob_prefix: Some(blob_prefix.clone() + "{{key}}"),
+            blob_append_uuid: Some(false),
+            batch: BatchConfig {
+                max_bytes: Some(1010),
+                ..config.batch
+            },
+            ..config
+        };
+        let sink = config.to_sink();
+        let groups = 3;
+        let (lines, input) = random_lines_with_stream_with_group_key(100, 30, groups);
+
+        sink.run(input).await.expect("Failed to run sink");
+
+        let blobs = config.get_blobs(blob_prefix.as_str()).await;
+        assert_eq!(blobs.len(), 3);
+        let response = stream::iter(blobs)
+            .fold(Vec::new(), |mut acc, blob| async {
+                let (_, lines) = config.get_blob(blob).await;
+                acc.push(lines);
+                acc
+            })
+            .await;
+
+        for i in 0..groups {
+            assert_eq!(&lines[(i * 10)..((i + 1) * 10)], response[i].as_slice());
+        }
+    }
+
+    impl AzureBlobSinkConfig {
+        pub async fn new_emulator() -> AzureBlobSinkConfig {
+            let config = AzureBlobSinkConfig {
+                connection_string: String::from("UseDevelopmentStorage=true;DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;"),
+                container_name: "logs".to_string(),
+                blob_prefix: None,
+                blob_time_format: None,
+                blob_append_uuid: None,
+                encoding: Encoding::Text.into(),
+                compression: Compression::None,
+                batch: Default::default(),
+                request: TowerRequestConfig::default(),
+            };
+
+            config.ensure_container().await;
+
+            config
+        }
+
+        pub fn to_sink(&self) -> VectorSink {
+            let cx = SinkContext::new_test();
+            let client = self.create_client().expect("Failed to create client");
+
+            self.new(client, cx).expect("Failed to create sink")
+        }
+
+        pub async fn get_blobs(&self, prefix: &str) -> Vec<String> {
```

Comment:
I would recommend naming this something obviously different from `get_blob`, like the underlying `list_blobs`.